compiler/src/dmd/dsymbolsem.d

```diff
@@ -6055,6 +6055,9 @@ void templateInstanceSemantic(TemplateInstance tempinst, Scope* sc, Expressions*
             continue;
         Type t = isType((*tempinst.tiargs)[i]);
         assert(t);
+        if (t.ty != TY.Tclass || t.ty != TY.Tstruct)
+            tempinst.error("argument for template this parameter must be a class or struct");
```

Comment:
Possibly missing syntax highlighting for class and struct. 
```suggestion
            tempinst.error("argument for template this parameter must be a `class` or `struct`");
```

Reply:
I have removed this error as I don't know why it triggers with `core.sync.condition.Condition`.

Reply:
Probably should be a '&&' condition
-if (t.ty != TY.Tclass || t.ty != TY.Tstruct)
+if (t.ty != TY.Tclass && t.ty != TY.Tstruct)

Reply:
@apz28 Thanks, looks good. But I decided to leave that check out of this pull.